src/hotspot/share/gc/serial/markSweep.cpp

```diff
@@ -159,9 +151,9 @@ void MarkSweep::preserve_mark(oop obj, markWord mark) {
   // sufficient space for the marks we need to preserve but if it isn't we fall
   // back to using Stacks to keep track of the overflow.
   if (_preserved_count < _preserved_count_max) {
-    _preserved_marks[_preserved_count++] = PreservedMark(obj, mark);
+    _preserved_marks[_preserved_count++] = OopAndMarkWord(obj, mark);
   } else {
-    _preserved_overflow_stack.push(PreservedMark(obj, mark));
+    _preserved_overflow_stack_set.get()->push_if_necessary(obj, mark);
```

Comment:
Why not `push_always`? I'd expect this mark must be preserved.

Reply:
Fixed.